analyzer/restapi-plugin/src/main/java/eu/fasten/analyzer/restapiplugin/mvn/api/impl/DebianResolutionApiServiceImpl.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package eu.fasten.analyzer.restapiplugin.mvn.api.impl;
+
+import eu.fasten.analyzer.restapiplugin.mvn.KnowledgeBaseConnector;
+import eu.fasten.analyzer.restapiplugin.mvn.api.DebianResolutionApiService;
+import eu.fasten.core.dependents.GraphResolver;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+
+@Lazy
+@Service
+public class DebianResolutionApiServiceImpl implements DebianResolutionApiService {
+
+    private static final Logger logger = LoggerFactory.getLogger(DebianResolutionApiServiceImpl.class);
+    private GraphResolver graphDebianResolver;
+
+    public DebianResolutionApiServiceImpl() {
+        try {
+            var graphDebianResolver = new GraphResolver();
+            graphDebianResolver.buildDependencyGraph(KnowledgeBaseConnector.dbCContext, KnowledgeBaseConnector.dependencyDebianGraphPath);
+            this.graphDebianResolver = graphDebianResolver;
+        } catch (Exception e) {
+            logger.error("Error constructing Debian dependency graph resolver", e);
+            System.exit(1);
```

Comment:
Why do you stop the REST API if an exception happens? Perhaps you can return a custom message to the user and show that an error happened when resolving dependents. That is, something like this `return ResponseEntity<String>(msg)`

Reply:
Ι stopped the REST-API because I followed the same logic with that of [ResolutionAPIService class](https://github.com/fasten-project/fasten/blob/ba6d41e64c627b04be8e523782fa2f28a50b7554/analyzer/restapi-plugin/src/main/java/eu/fasten/analyzer/restapiplugin/mvn/api/impl/ResolutionApiServiceImpl.java#L65) but if its required I can change it to the proposed return message.